docs/docs/reference/plugin-definition-syntax.md

```diff
@@ -474,11 +474,13 @@ settings:
   env: SOME_API_KEY
 ```
 
-<br/>
-
 :::caution
-  <p><code>kind: hidden</code> is deprecated in favour of <a href="#settingshidden"><code>hidden</code></a>.</p>
-  <p><code>kind: password</code> is deprecated in favour of <a href="#settingssensitive"><code>sensitive</code></a>.</p>
+  <p><code>kind: hidden</code> is deprecated in favour of <a href="#settingshidden"><code>hidden: true</code></a>.</p>
+  <p><code>kind: password</code> is deprecated in favour of <a href="#settingssensitive"><code>sensitive: true</code></a>.</p>
+:::
+
+:::tip
+  <p>Starting with Meltano v3.7, settings of kind `date_iso8601` can be relative dates, like `3 days ago`, `yesterday`, `last week`, etc.</p>
```

Comment:
This is my first exposure to this issue/PR, so maybe this has been discussed already, but it's rather confusing that the setting is named `date_iso8601`, yet accepts things that are not in an ISO 8601 format. Would it be worth introducing a new setting type named `datetime` or something like that, and making it accept a superset of what `date_iso8601` accepts?

Then again, since the input is normalized into an ISO 8601 format before it reaches the plugin for which the setting is for, maybe this is fine. The setting kind can be thought of as the kind of data that the plugin accepts, rather than what Meltano will accept for it.

Reply:
> The setting kind can be thought of as the kind of data that the plugin accepts, rather than what Meltano will accept for it.

That's my understanding of it as well.

> Would it be worth introducing a new setting type named `datetime` or something like that, and making it accept a superset of what `date_iso8601` accepts?

Maybe, but then we'd need to bulk-update plugins in the Hub to have a `start_date` of type `datetime` in order for they to support this, or at least users have to override the setting in their `meltano.yml`.